src/components/Modals/NewCourse/NewCourseModal.vue

```diff
@@ -109,6 +110,10 @@ export default Vue.extend({
   methods: {
     selectCourse(result: MatchingCourseSearchResult) {
       this.selectedCourse = result;
+      if (this.isSelfCheck) {
+        this.addCourse();
+        return;
+      }
```

Comment:
This is a smart hack, but I think it's a better idea to create a specialized new course modal (e.g. `<add-course-for-self-check-modal />`) that just let you choose courses and semester. 

This can be easily achieved by copy-pasting all the code in `NewCourseModal` and then remove the code you don't need (in this case it's mostly the requirement editor code). This approach will create some duplicated code, but I think it's worth it, because then the `NewCourseModal` don't need to worry about editing semester at all.